src/app/user/edit-profile/change-password-form/change-password-form.component.ts

```diff
@@ -68,13 +68,23 @@ export class ChangePasswordFormComponent implements OnInit {
 
     this.auth
       .loginUser(updatedUser.username, this.currentPassword.value)
-      .pipe(take(1))
+      .pipe(
+        take(1),
+        switchMap((valid) => {
+          if (valid) {
+            this.userService.updateUser(updatedUser).pipe(take(1)).subscribe();
```

Comment:
This is roughly the same problem. We can't subscribe to observables inside of another observable. SwitchMap is intended to return the result of another observable. So in this case you can return the update call instead of subscribing to it. Then it's available in a later pipe so you could react to the successful or not successful response of this. Alternatively, another approach is to create your own observable with a Subject or Behavior Subject and "next(...)" onto that stream to do more side effect processing. Meaning... if you don't care about the result of update user but you want to make sure it happens, you can create your own stream and this code would just tell that stream to do whatever it needs to do. (fire and forget).

Let me know if you want to talk about any of these concepts.

Reply:
Here's an example of switchmap in a similar scenaro:

``` typescript
this.authenticated = this.authService.isAuthenticated()
      .pipe(
        take(1),
        switchMap(isAuthenticated => {
          if (isAuthenticated) {
            return this.apiService.isUserLoggedIn()
              .pipe(
                catchError(error => {
                  if (error instanceof HttpErrorResponse && error.status === 401) {
                    this.authService.removeAllTokens();
                    this.authService.redirect();
                  }
                  return of(false);
                })
              );
          }

          return of(false);
        }),
        tap(() => this.loggingService.logEvent('Logged in.'))
      );
```